source/exe/posix/platform_checks.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <fcntl.h>
+#include <sys/resource.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#include "envoy/filesystem/filesystem.h"
+
+#include "common/common/assert.h"
+
+namespace Envoy {
+
+static void check_sysctl_long(std::string path, long expected, Filesystem::Instance& file_system) {
+  if (file_system.fileExists(path)) {
+    long current = std::stol(file_system.fileReadToEnd(path));
```

Comment:
you need to handle failures in two ways:

1. What happens if fileReadToEnd fails?
2. What happens if the string returned from that is not convertible to an int64_t.